### PR TITLE
Add 1-D and 3-D pooling operations and fix handling of incomplete attributes

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/max_pool_nd.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    : Node(
+          ir::OpKind(MaxPoolNdSymbol(spatial_dim_count)), {input},
+          [&]() {
+            return NodeOutputShape(input, spatial_dim_count, kernel_size,
+                                   stride, padding);
+          },
+          /*num_outputs=*/1,
+          xla::util::MHash(spatial_dim_count, kernel_size, stride, padding)),
+      spatial_dim_count_(spatial_dim_count),
+      kernel_size_(kernel_size.begin(), kernel_size.end()),
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/max_pool_nd.cpp]: ?
std::move to all.

File: torch_xla/csrc/ops/max_pool_nd_backward.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(
+    const Value& grad_output, const Value& input, xla::int64 spatial_dim_count,
+    tensorflow::gtl::ArraySlice<const xla::int64> kernel_size,
+    tensorflow::gtl::ArraySlice<const xla::int64> stride,
+    tensorflow::gtl::ArraySlice<const xla::int64> padding) {
+  auto lower_for_shape_fn =
+      [spatial_dim_count, kernel_size, stride,
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/max_pool_nd_backward.cpp]: Ditto

File: torch_xla/csrc/tensor_methods.cpp [hunk 3]
```diff
@@ -113,6 +113,29 @@ void CheckDimensionSize(const XLATensor& t, xla::int64 dim,
...
+  if (list.empty()) {
+    result = std::move(def);
+  } else {
+    result = xla::util::ToVector<xla::int64>(list);
+  }
+  if (result.size() == 1 && length > 1) {
+    result.resize(length, result[0]);
+    return result;
+  }
+  if (result.size() != length) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: XLA_CHECK_EQ() << ....

File: torch_xla/csrc/ops/max_pool_nd.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/ops/max_pool_nd.cpp]: Leftover from when the parameters were slices, probably. Done.

File: torch_xla/csrc/ops/avg_pool_nd.cpp [hunk 4]
```diff
@@ -12,51 +12,68 @@ namespace {
...
 xla::Shape NodeOutputShape(
-    const Value& input,
+    const Value& input, xla::int64 spatial_dim_count,
     tensorflow::gtl::ArraySlice<const xla::int64> kernel_size,
     tensorflow::gtl::ArraySlice<const xla::int64> stride,
     tensorflow::gtl::ArraySlice<const xla::int64> padding,
     bool count_include_pad) {
   auto lower_for_shape_fn =
-      [stride, padding, kernel_size, count_include_pad](
+      [spatial_dim_count, kernel_size, stride, padding, count_include_pad](
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/ops/avg_pool_nd.cpp]: Done.

File: torch_xla/csrc/ops/avg_pool_nd_backward.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+namespace {
+
+xla::Shape NodeOutputShape(
+    const Value& grad_output, const Value& input, xla::int64 spatial_dim_count,
+    tensorflow::gtl::ArraySlice<const xla::int64> kernel_size,
+    tensorflow::gtl::ArraySlice<const xla::int64> stride,
+    tensorflow::gtl::ArraySlice<const xla::int64> padding,
+    bool count_include_pad) {
+  auto lower_for_shape_fn =
+      [spatial_dim_count, kernel_size, stride, padding, count_include_pad](
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/ops/avg_pool_nd_backward.cpp]: Done.

File: torch_xla/csrc/ops/max_pool_nd.cpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+namespace {
+
+// Infers the output shape of the max pooling operation.
+xla::Shape NodeOutputShape(
+    const Value& input, xla::int64 spatial_dim_count,
+    tensorflow::gtl::ArraySlice<const xla::int64> kernel_size,
+    tensorflow::gtl::ArraySlice<const xla::int64> stride,
+    tensorflow::gtl::ArraySlice<const xla::int64> padding) {
+  auto lower_for_shape_fn =
+      [spatial_dim_count, kernel_size, stride,
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/ops/max_pool_nd.cpp]: Done.

File: torch_xla/csrc/ops/max_pool_nd_backward.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/max_pool_nd_backward.cpp]: Done.

File: torch_xla/csrc/tensor_methods.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Done.

File: torch_xla/csrc/tensor_methods.cpp [hunk 7]
```diff
@@ -113,6 +113,28 @@ void CheckDimensionSize(const XLATensor& t, xla::int64 dim,
...
+  if (list.empty()) {
+    result = std::move(def);
+  } else {
+    result = xla::util::ToVector<xla::int64>(list);
+  }
+  if (result.size() == 1 && length > 1) {
+    result.resize(length, result[0]);
+    return result;
+  }
+  XLA_CHECK_EQ(result.size(), length)
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: You do not need to repeat streaming the two values, as those are part of the CHECK message. You stream the name and some other textual context and you are done.

Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Done.

**Pull request merged**